modules/social_features/social_activity/modules/social_activity_filter/src/Form/FilterSettingsForm.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Drupal\social_activity_filter\Form;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Extension\ModuleHandlerInterface;
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\node\Entity\NodeType;
+use Drupal\taxonomy\Entity\Vocabulary;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class FilterSettingsForm.
+ *
+ * @package Drupal\unpd_cop\Form
+ */
+class FilterSettingsForm extends ConfigFormBase implements ContainerInjectionInterface {
+
+  /**
+   * The Module Handler.
+   *
+   * @var \Drupal\Core\Extension\ModuleHandlerInterface
+   */
+  protected $moduleHandler;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(ConfigFactoryInterface $config_factory, ModuleHandlerInterface $moduleHandler) {
+    parent::__construct($config_factory);
+    $this->moduleHandler = $moduleHandler;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('module_handler')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'social_activity_filter_settings';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getEditableConfigNames() {
+    return ['social_activity_filter.settings'];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+
+    // Get the configuration file.
+    $config = $this->config('social_activity_filter.settings');
+    $taxonomyFields = $config->get('taxonomy_fields');
+
+    $form['social_activity_filter'] = [
+      '#type' => 'fieldset',
+      '#title' => $this->t('Filter options list'),
+    ];
+
+    $vocabulariesList = [];
+
+    /** @var \Drupal\taxonomy\Entity\Vocabulary */
+    foreach (Vocabulary::loadMultiple() as $vid => $vocabulary) {
+      $referencedField = isset($taxonomyFields[$vid]) ? $taxonomyFields[$vid] : $this->t('none');
+      $vocabulariesList[$vid] = $vocabulary->get('name') . " (field: $referencedField)";
+    }
+
+    $form['social_activity_filter']['vocabulary'] = [
```

Comment:
@volodymyr-sydor-lemberg Please add a description which explains that field will be added automatically.
What will happen if both content types use the same taxonomy but with different fields?

Reply:
@ribel I've added description for those cases

Reply:
done